pkg/hardware/config.go

```diff
@@ -88,6 +98,74 @@ func (hc *HardwareConfig) ValidateBmcRefMapping() error {
 	return nil
 }
 
+func (hc *HardwareConfig) ValidateBMC() error {
+	secretRefMap := hc.initSecretRefMap()
+	bmcIpMap := make(map[string]struct{}, len(hc.bmcList))
+	for _, bmc := range hc.bmcList {
+		if bmc.Name == "" {
+			return fmt.Errorf("bmc name is required")
+		}
+
+		if bmc.Spec.AuthSecretRef.Name == "" {
+			return fmt.Errorf("authSecretRef name required for bmc %s", bmc.Name)
+		}
+
+		if bmc.Spec.AuthSecretRef.Namespace != constants.EksaSystemNamespace {
+			return fmt.Errorf("invalid authSecretRef namespace: %s for bmc %s", bmc.Spec.AuthSecretRef.Namespace, bmc.Name)
+		}
+
+		if _, ok := secretRefMap[bmc.Spec.AuthSecretRef.Name]; !ok {
+			return fmt.Errorf("bmc authSecretRef: %s not present in hardware config", bmc.Spec.AuthSecretRef.String())
+		}
+
+		if _, ok := bmcIpMap[bmc.Spec.Host]; ok {
+			return fmt.Errorf("duplicate host IP: %s for bmc %s", bmc.Spec.Host, bmc.Name)
+		} else {
+			bmcIpMap[bmc.Spec.Host] = struct{}{}
+		}
+
+		if err := networkutils.ValidateIP(bmc.Spec.Host); err != nil {
+			return fmt.Errorf("bmc host IP: %v", err)
+		}
+
+		if bmc.Spec.Vendor == "" {
+			return fmt.Errorf("bmc: %s vendor is required", bmc.Name)
+		}
+	}
+
+	return nil
+}
+
+func (hc *HardwareConfig) ValidateBmcSecretRefs() error {
+	for _, s := range hc.secretList {
+		if s.Name == "" {
+			return fmt.Errorf("secret name is required")
+		}
+		if s.Namespace != constants.EksaSystemNamespace {
+			return fmt.Errorf("invalid secret namespace: %s for secret: %s expected: %s", s.Namespace, s.Name, constants.EksaSystemNamespace)
+		}
+		du, dOk := s.Data["username"]
```

Comment:
Overall PR looks good to me! Just one thing. Can you pl make variable names more readable?